staticcheck/util/directive.go

```diff
@@ -0,0 +1,66 @@
+package util
+
+import (
+	"go/token"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+	"honnef.co/go/tools/analysis/lint"
+)
+
+type ignore interface {
+	match(pos token.Position) bool
+}
+
+type lineIgnore struct {
+	File string
+	Line int
+}
+
+func (li lineIgnore) match(pos token.Position) bool {
+	return pos.Filename == li.File && pos.Line == li.Line
+}
+
+type fileIgnore struct {
+	File string
+}
+
+func (fi fileIgnore) match(pos token.Position) bool {
+	return pos.Filename == fi.File
+}
+
+func asIgnores(fs *token.FileSet, category string, lds []lint.Directive) []ignore {
+	var igs []ignore
+	for _, ld := range lds {
+		for _, c := range strings.Split(ld.Arguments[0], ",") {
+			if m, _ := filepath.Match(c, category); m {
```

Comment:
It's not clear to me how `category` is used here.

It seems like the actual value passed from the tests and util.go are the analyzers' name. In staticcheck they should be something like `ST0001`?

- Could we rename it to something like `analyzerName`?

- Unclear to me at first glance why `filepath.Match` is being used against category. It seems like the glob matching `ST* -> ST1000` is not documented on staticcheck website. It would be nice if we could add a tiny comment either here or on top of the func to either explain the behavior or to reference the original implementation.

Nit: I would appreciate it if you could add a comment, either on top of each func or on top of the file, pointing to a permalink from staticcheck. Such as https://github.com/dominikh/go-tools/blob/4ec1f474ca6c0feb8e10a8fcca4ab95f5b5b9881/lintcmd/directives.go#L9. It will help us diff against a future version of staticcheck to understand how things change for easier maintenance.

Reply:
Done!